spark/src/test/scala/org/apache/hadoop/fs/versioned/VersionedFileSystemSpec.scala

```diff
@@ -0,0 +1,73 @@
+package org.apache.hadoop.fs.versioned
+
+import com.gu.tableversions.core.Partition
+import com.gu.tableversions.core.Partition.{ColumnValue, PartitionColumn}
+import com.gu.tableversions.metastore.VersionPaths
+import com.gu.tableversions.spark.SparkHiveSuite
+import org.apache.spark.sql.SaveMode
+import org.scalatest.{FlatSpec, Matchers}
+
+class VersionedFileSystemSpec
+    extends FlatSpec
+    with Matchers
+    with SparkHiveSuite
+    with S3TestSuite
+    with S3SparkTestSuite {
+
+  spark.sparkContext.setLogLevel("ERROR")
+
+  "flattening List[Partition]" should "produce configuration entries" in {
```

Comment:
This test should go to a spec for wherever the Hadoop conversion functions end up.